framework/audio/engine/internal/audioengine.cpp

```diff
@@ -0,0 +1,197 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "audioengine.h"
+
+#include <soloud.h>
+
+#include "log.h"
+#include "ptrutils.h"
+
+#include "modularity/ioc.h"
+#include "iaudiodriver.h"
+
+using namespace mu::audio::engine;
+
+constexpr int BUF_SIZE{ 1024 };
+
+struct AudioEngine::SL {
+    SoLoud::Soloud engine;
+};
+
+AudioEngine::AudioEngine()
+{
+    m_sl = std::unique_ptr<SL>(new SL);
+}
+
+AudioEngine::~AudioEngine()
```

Comment:
Not needed, because already implemented in the interface

Reply:
removed